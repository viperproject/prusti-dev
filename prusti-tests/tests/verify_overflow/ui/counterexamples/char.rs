// compile-flags: -Punsafe_core_proof=true -Pcounterexample=true

use prusti_contracts::*;

#[ensures(!result)]
fn test1(x: char) -> bool {
    x == 'c'
}

fn main() {}
