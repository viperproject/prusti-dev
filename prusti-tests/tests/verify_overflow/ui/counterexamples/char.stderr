error: [Prusti: verification error] postcondition might not hold.
 --> $DIR/char.rs:5:11
  |
5 | #[ensures(!result)]
  |           ^^^^^^^
  |
note: the error originates here
 --> $DIR/char.rs:6:1
  |
6 | / fn test1(x: char) -> bool {
7 | |     x == 'c'
8 | | }
  | |_^
note: counterexample for "x"
       value:   'c' (0x63)
 --> $DIR/char.rs:6:10
  |
6 | fn test1(x: char) -> bool {
  |          ^
note: counterexample for "result"
       value:   true
 --> $DIR/char.rs:7:5
  |
7 |     x == 'c'
  |     ^^^^^^^^

error: aborting due to previous error

